### PR TITLE
Add 3rd order spherical harmonics

### PR DESCRIPTION
**Pull request by po09i**
## Description
- This PR adds the possibility to use the 3rd order spherical harmonics. 
- It adds the Terra and updates "Investigational_Device_7T" constraints 
- It adds rmse to the list of available optimization criteria

### CODE REVIEW

File: shimmingtoolbox/coils/spher_harm_basis.py [hunk 1]
```diff
@@ -522,32 +513,52 @@ def _check_basis_inputs(x, y, z, orders):
...
+    Args:
+        order (int): Order of the spherical harmonics
+        manufacturer (str): Manufacturer of the scanner.
+
+    Returns:
+
+    """
+    if manufacturer == 'Siemens' and order == 3:
+        return 4
+    return 2 * order + 1
```
Comment (4rnaudB) [inline shimmingtoolbox/coils/spher_harm_basis.py]: Do other manufacturers have all the coefficients for the third order?

File: shimmingtoolbox/coils/spher_harm_basis.py [hunk 2]
```diff
@@ -556,33 +567,34 @@ def get_flip_matrix(shim_cs='ras', orders=(1, 2), manufacturer=None):
...
-    out_matrix = np.array(temp_list_out)
+    # Y, Z, X, XY, ZY, Z2, ZX, X2 - Y2, Y(X2 - Y2), XYZ, Z2Y, Z3, Z2X, Z(X2 - Y2), X(X2 - Y2)
+    out_dict = {}
+    for order in orders:
+        if order == 1:
+            out_dict[1] = np.array([xyz_cs[1], xyz_cs[2], xyz_cs[0]])
+        if order == 2:
+            out_dict[2] = np.array([xyz_cs[0] * xyz_cs[1], xyz_cs[2] * xyz_cs[1], 1, xyz_cs[2] * xyz_cs[0], 1])
+        if order == 3:
+            # TODO: Verify
```
Comment (4rnaudB) [inline shimmingtoolbox/coils/spher_harm_basis.py]: Done?

File: shimmingtoolbox/coils/spher_harm_basis.py [hunk 3]
```diff
@@ -556,33 +567,34 @@ def get_flip_matrix(shim_cs='ras', orders=(1, 2), manufacturer=None):
...
         manufacturer = manufacturer.upper()
-    if manufacturer == 'SIEMENS':
-        out_matrix = _reorder_to_siemens(out_matrix, orders)
-    elif manufacturer == 'GE':
-        out_matrix = _reorder_to_ge(out_matrix, orders)
-        out_matrix = _reorder_to_ge2(out_matrix, orders)
-    elif manufacturer == 'PHILIPS':
-        out_matrix = _reorder_to_philips(out_matrix, orders)
+
+    if manufacturer in ['SIEMENS', 'GE', 'PHILIPS']:
```
Comment (4rnaudB) [inline shimmingtoolbox/coils/spher_harm_basis.py]: Not a variable?

File: shimmingtoolbox/conversion.py [hunk 4]
```diff
@@ -3,7 +3,56 @@
...
+    Returns:
+        Float or array of floats converted
+    """
+    if prefix_in not in METRIC_PREFIXES or prefix_out not in METRIC_PREFIXES:
+        raise ValueError('Invalid prefix')
+    factor = (METRIC_PREFIXES[prefix_in] / METRIC_PREFIXES[prefix_out]) ** power
+    return x * factor
+
+
+def unit_per_metric_unit_to_unit_per_metric_unit(x, prefix_in, prefix_out, power=1):
```
Comment (4rnaudB) [inline shimmingtoolbox/conversion.py]: Could be merged with function above

Comment (po09i) [reply shimmingtoolbox/conversion.py]: Fixed: [ee800e6](https://github.com/shimming-toolbox/shimming-toolbox/pull/497/commits/ee800e6c2c4ec79a97e5c5e0faf409cdbf66d708)

File: shimmingtoolbox/coils/spher_harm_basis.py (hunk 1 above)
Comment (po09i) [reply shimmingtoolbox/coils/spher_harm_basis.py]: I'm not sure, so for now I assume they have all channels

**Review by 4rnaudB — COMMENTED** [review]
The code looks good. I compared it with my 3rd order and there is a slight performance difference. The newly implemented 3rd order nearly changed anything in the optimization compared to the second order while coil fake coil 3rd order helped more.

**Comment by po09i** [comment]
Siemens only uses 4 channels for the 3rd order (at least for now on the Terra), maybe that might be why the ones you have (which have all 7 channels for the 3rd order) give a better improvement.

**Review by 4rnaudB — APPROVED** [review]

**Pull request merged**